### ISSUE TITLE
[Spaces] Rooms filter: search rooms from all Spaces and orphaned, not only from selected Space

### ISSUE THREAD

**Issue by MurzNN** [vector-im/element-web#16683]
**Is your suggestion related to a problem? Please describe.**
When I happily organized (split) my long list of rooms via brand new "Spaces", there become very hard to quickly find needed room!

Because now I need at first to remember the space to which the needed room is attached, switch to it, and only after this - can filter list of rooms via "Filter" input field under the rooms list, to find needed one.

Also for find the DM room (someone from "People" group), now I must switch to "Home" (orphaned rooms) first, that is additional action too.

**Describe the solution you'd like**
Will be good to extend rooms filter functionality to show results from all spaces, not only from selected (or only orphaned rooms), especially, when no results are found. 

The good quick solution is to show the plain mixed list of all matched rooms from all Spaces.

Better solution can be show per-space groups of rooms in search results of filter, with current Space (or orphaned rooms if no Space is selected) at first place, and other matched Spaces groups below.

And, maybe, show the matched Spaces names too (as clickable result) in filter results, to use the same filter for quickly find needed Space by name too?

**Comment by SimonBrandner** [comment]
Personally, I'd prefer something like what is suggested in [#1854](https://github.com/vector-im/element-web/issues/1854). Discord has this already:
![Screenshot_20210315_110333](https://user-images.githubusercontent.com/25768714/111136490-1800a100-857e-11eb-9c38-423d37f65495.png)

**Comment by t3chguy** [comment]
Closing in favour of https://github.com/vector-im/element-web/issues/16998
